src/behavior-considered-undefined.md

```diff
@@ -58,7 +58,11 @@ Please read the [Rustonomicon] before writing unsafe code.
 * Invoking undefined behavior via compiler intrinsics.
 * Executing code compiled with platform features that the current platform
   does not support (see [`target_feature`]), *except* if the platform explicitly documents this to be safe.
-* Calling a function with the wrong call ABI or unwinding from a function with the wrong unwind ABI.
+* Calling a function with the wrong [call ABI][abi].
+* Unwinding, or catching an unwind, [inappropriately][unwinding].
+* Deallocating a Rust stack frame without executing destructors
+  for local variables owned by the stack frame. This can occur
+  with C functions like `longjmp`.
```

Comment:
Can this be made part of "unwinding inappropriately"? We could say that from a Rust perspective, what `longjmp` does is a form of unwinding, thus avoiding the need to make this a completely separate kind of UB.

This item feels a bit strange because it doesn't even refer to an operation that you can perform in Rust. It is, in a sense, part of the larger class of UB caused by "breaking basic assumptions about the CPU/machine state that the compiler relies on" (like messing up the page table or the floating point rounding mode or whatever).

Reply:
But `longjmp` is specifically _not_ a form of unwinding, from a Rust perspective (despite the "forced unwinding" implementation). They're both forms of non-local control flow, so I could potentially group them that way, but I'm not sure that would be clearer to very many people.

Can't someone effectively `longjmp` or otherwise modify the stack pointer with assembly in Rust? If not, you're right, this isn't something you can do "in Rust," per se; but it isn't inherently UB due to the way Rust is specified, and we'd _like_ to make well-defined in some cases. Additionally, as far as I know, it's not currently documented as UB elsewhere (other than in the `c-unwind` RFC), and there's nowhere better for it to go.

Reply:
Inline assembly may only do things that either Rust cannot observe, or that you could do in Rust as well.

We should probably have a section somewhere with an incomplete list of extra UB to be aware of when mixing Rust with other languages (including inline assembly). But this is qualitatively different UB so it should be clearly separate from UB that is declared in the Rust Abstract Machine.